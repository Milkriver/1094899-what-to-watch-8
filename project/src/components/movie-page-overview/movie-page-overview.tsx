import movies from '../../mocks/movies';

export function MoviePageOverview(): JSX.Element {
  return (
    <>
      <div className="film-rating">
        <div className="film-rating__score">8,9</div>
        <p className="film-rating__meta">
          <span className="film-rating__level">Very good</span>
          <span className="film-rating__count">240 ratings</span>
        </p>
      </div>
      <div className="film-card__text">
        <p>{movies[0].description}</p>
        <p className="film-card__director"><strong>Director: {movies[0].director}</strong></p>
        <p className="film-card__starring"><strong>Starring: {movies[0].starring}</strong></p>
      </div>
    </>
  );
}
