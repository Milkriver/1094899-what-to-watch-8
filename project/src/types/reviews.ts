export interface IReview {
    id: number,
    movieId: number,
    author: string,
    date: string,
    rating: number,
    text: string,
  }
