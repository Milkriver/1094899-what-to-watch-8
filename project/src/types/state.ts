import { AuthorizationStatus } from '../const';
import { IMovie } from './common';
import { IReviewResponse } from './reviews';

export interface IState {
  currentGenre: string
  movies: IMovie[]
  isDataLoaded: boolean
  authorizationStatus: AuthorizationStatus,
  movie: IMovie,
  sameMovies: IMovie[],
  reviews: IReviewResponse[],
  promoMovie: IMovie,
}
