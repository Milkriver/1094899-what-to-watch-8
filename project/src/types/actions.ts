import { AxiosInstance } from 'axios';
import { Action } from 'redux';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { IState } from './state';

export enum ActionType {
  ChangeGenre = 'movieList/ChangeGenre',
  LoadMovies = 'movieList/LoadMovies',
  LoadSingleMovie = 'movieCard/LoadSingleMovie',
  LoadActiveMovie = 'movieCard/LoadActiveMovie',
  LoadSameGenreMovies = 'movieCard/LoadSameGenreMovies',
  LoadPromoMovie = 'movieCard/LoadPromoMovie',
  LoadFavouriteMovies = 'userList/LoadFavouriteMovies',
  AddFavoriteMovie = 'userList/AddFavoriteMovie',
  RemoveFavoriteMovie = 'userList/RemoveFavoriteMovie',
  LoadReviews = 'movieCard/LoadReviews',
  AddReview = 'movieCard/AddReview',
  RequireAuthorization = 'user/requireAuthorization',
  RequireLogout = 'user/requireLogout',
  LoadUserData = 'user/loadUserData',
}

export type ThunkActionResult<R = Promise<void>> = ThunkAction<R, IState, AxiosInstance, Action>;

export type ThunkAppDispatch = ThunkDispatch<IState, AxiosInstance, Action>;
