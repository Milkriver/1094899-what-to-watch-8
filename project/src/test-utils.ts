import thunk, { ThunkDispatch } from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';
import { createAPI } from './services/api';
import { configureMockStore, MockStoreCreator } from '@jedmao/redux-mock-store';
import { IState } from './types/state';
import { Action } from 'redux';
import { AxiosInstance } from 'axios';

interface IFakeInfrastructure {
  mockAPI: MockAdapter
  mockStore: MockStoreCreator<
    IState,
    Action,
    ThunkDispatch<IState, AxiosInstance, Action>
  >
}

export const makeFakeInfrastructure = (): IFakeInfrastructure => {
  const onFakeUnauthorized = jest.fn();
  const api = createAPI(onFakeUnauthorized());
  const mockAPI = new MockAdapter(api);
  const middlewares = [thunk.withExtraArgument(api)];

  const mockStore: MockStoreCreator<
    IState,
    Action,
    ThunkDispatch<IState, AxiosInstance, Action>
  > = configureMockStore(middlewares);

  mockAPI
    .onAny()
    .reply(200, {});

  return {
    mockAPI,
    mockStore,
  };
};
