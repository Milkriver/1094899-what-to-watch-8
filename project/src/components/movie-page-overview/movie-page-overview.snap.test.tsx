import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { makeFakeInfrastructure } from '../../test-utils';
import { IMovie } from '../../types/common';
import { MoviePageOverview } from './movie-page-overview';


const MOCK_MOVIE_1: IMovie = { id: 1, name: '', posterImage: '', previewImage: '', backgroundImage: '', backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 1, scoresCount: 1, director: '', starring: ['1', '2'], runTime: 1, genre: '', released: 1, isFavorite: true };
describe('Component: MoviePageOverview', () => {
  const { mockStore } = makeFakeInfrastructure();

  const store = mockStore({
    MOVIES: {
      movie: {
        id: 1, name: '', posterImage: '', previewImage: '', backgroundImage: '',
        backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 1, scoresCount: 1, director: '', starring: ['', ''],
        runTime: 1, genre: '', released: 1, isFavorite: true,
      },
    },
    USER: {},
    REVIEWS: {},
  });

  test('should render correctly', () => {
    const history = createMemoryHistory();
    const page = (
      <Provider store={store}>
        <Router history={history}>
          <MoviePageOverview
            movie={MOCK_MOVIE_1}
          />;
        </Router>
      </Provider>
    );
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
