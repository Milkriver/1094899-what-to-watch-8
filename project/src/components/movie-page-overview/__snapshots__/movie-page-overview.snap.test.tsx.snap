// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MoviePageOverview should render correctly 1`] = `
<div>
  <div
    class="film-rating"
  >
    <div
      class="film-rating__score"
    >
      1
    </div>
    <p
      class="film-rating__meta"
    >
      <span
        class="film-rating__level"
      >
        Bad
      </span>
      <span
        class="film-rating__count"
      >
        1
      </span>
    </p>
  </div>
  <div
    class="film-card__text"
  >
    <p />
    <p
      class="film-card__director"
    >
      <strong>
        Director: 
        
      </strong>
    </p>
    <p
      class="film-card__starring"
    >
      <strong>
        Starring: 
        1
        2
      </strong>
    </p>
  </div>
  ;
</div>
`;
